broker-auth-local/src/main/java/org/aktin/broker/auth/apikey/ApiKeyManagementEndpoint.java

```diff
@@ -0,0 +1,150 @@
+import org.aktin.broker.rest.Authenticated;
+import org.aktin.broker.rest.RequireAdmin;
+
+/**
+ * RESTful endpoint for managing API keys. This class provides operations to retrieve, create, activate, and deactivate API keys. Access to these
+ * operations requires authentication and admin privileges.
+ *
+ * @author akombeiz@ukaachen.de
+ */
+@Authenticated
+@RequireAdmin
+@Path("api-keys")
+public class ApiKeyManagementEndpoint {
+
+  private static final Logger log = Logger.getLogger(ApiKeyManagementEndpoint.class.getName());
+
+  @Inject
+  ApiKeyPropertiesAuthProvider authProvider;
+
+  /**
+   * Retrieves all API keys.
+   *
+   * @return A Response with {@code 200} containing all API keys as a string, or {@code 500} if retrieval fails.
+   */
+  @GET
+  public Response getApiKeys() {
+    try {
+      PropertyFileAPIKeys apiKeys = authProvider.getInstance();
+      Properties props = apiKeys.getProperties();
+      return Response.ok(convertPropertiesToString(props)).build();
+    } catch (IOException e) {
+      log.severe("Error retrieving API keys: " + e.getMessage());
+      return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
+
+  /**
+   * Converts Properties to a string representation.
+   *
+   * @param props The Properties object to convert.
+   * @return A string representation of the properties.
+   */
+  private String convertPropertiesToString(Properties props) {
+    StringBuilder response = new StringBuilder();
+    for (String name : props.stringPropertyNames()) {
+      response.append(name).append("=").append(props.getProperty(name)).append("\n");
+    }
+    return response.toString();
+  }
+
+  /**
+   * Creates a new API key.
+   *
+   * @param apiKeyDTO The DTO containing the API key and client DN.
+   * @return A Response indicating the result of the operation: <ul>
+   * <li>{@code 201} - API key created successfully</li>
+   * <li>{@code 400} - API key or client DN are not provided</li>
+   * <li>{@code 409} - API key already exists</li>
+   * <li>{@code 500} - When there is an error in saving the properties file or API keys instance is not initialized</li>
+   * </ul>
+   */
+  @POST
+  @Consumes(MediaType.APPLICATION_XML)
+  public Response postApiKey(ApiKeyDTO apiKeyDTO) {
+    String apiKey = apiKeyDTO.getApiKey();
+    String clientDn = apiKeyDTO.getClientDn();
+    if (apiKey == null || clientDn == null) {
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+    try {
+      authProvider.addNewApiKeyAndUpdatePropertiesFile(apiKey, clientDn);
+      return Response.status(Status.CREATED).build();
+    } catch (IllegalArgumentException e) {
+      return Response.status(Status.CONFLICT).build();
+    } catch (IOException e) {
+      return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
+
+  /**
+   * Activates an API key.
+   *
+   * @param apiKey The API key to activate.
+   * @return A Response indicating the result of the operation.
+   */
+  @PUT
```

Comment:
We don't PUT any resource on the server from REST point of view. Use POST instead

Reply:
I think PUT Is applicable, because we are modifying the state of an existing resource in an idempotent way as described here: https://restfulapi.net/rest-put-vs-post/

Reply:
A toggle method is not RESTful, because it is not idempotent